### PR TITLE
Keeping filter active after selecting - useful for MultiSelect

### PR DESCRIPTION
**Pull request by siredmar**
This PR adds the feature to have the filter settings stay active during selection.

See Issue #277.

### CODE REVIEW

**Comment by AlecAivazis** [comment]
Hey @siredmar! Thanks for the thorough contribution. I especially appreciate that you took the time to add documentation.  I see a few minor tweaks but this should be able to be merged relatively quickly

File: multiselect.go [hunk 1]
```diff
@@ -100,7 +100,9 @@ func (m *MultiSelect) OnChange(key rune, config *PromptConfig) {
 				// otherwise just invert the current value
 				m.checked[selectedOpt.Index] = !old
 			}
-			m.filter = ""
+			if config.FilterKeep == false {
```
Comment (AlecAivazis) [inline multiselect.go]: Small nitpick - can we change this to `if !config.FilterKeep`

Comment (siredmar) [reply multiselect.go]: I choose KeepFilter if that's okay with you.

Comment (AlecAivazis) [reply multiselect.go]: That sounds good to me! I'll approve this once you've updated all the things. Thanks again!

Comment (siredmar) [reply multiselect.go]: It was me a pleasure. I do need this feature myself, so it's just developed it.

**Pull request merged**